src/vs/workbench/services/files/node/fileService.ts

```diff
@@ -366,6 +366,34 @@ export class FileService implements IFileService {
 
 						// 5.) resolve
 						return this.resolve(resource);
+					}, error => {
+						// Can't use 'w' for hidden files, so truncate and use 'r+' if the file exists
+						if (!exists || error.code !== 'EPERM') {
```

Comment:
@seishun should we add a check for windows here? as far as I know, hidden files (and this issue) only manifest on windows